plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/AbstractTestIcebergSmoke.java

```diff
@@ -1616,4 +1639,81 @@ public void testOptimizedMetadataQueries()
 
         dropTable("test_metadata_optimization");
     }
+
+    @Test
+    public void testIncorrectIcebergFileSizes()
+            throws Exception
+    {
+        // Create a table with a single insert
+        assertUpdate("CREATE TABLE test_iceberg_file_size (x BIGINT) WITH (format='PARQUET')");
+        assertUpdate("INSERT INTO test_iceberg_file_size VALUES (123), (456), (758)", 3);
+
+        // Get manifest file
+        MaterializedResult result = computeActual("SELECT path FROM \"test_iceberg_file_size$manifests\"");
+        assertEquals(result.getRowCount(), 1);
+        String manifestFile = (String) result.getOnlyValue();
+
+        // Read manifest file
+        Schema schema;
+        GenericData.Record entry = null;
+        try (DataFileReader<GenericData.Record> dataFileReader = new DataFileReader<>(new File(manifestFile), new GenericDatumReader<>())) {
+            schema = dataFileReader.getSchema();
+            int recordCount = 0;
+            while (dataFileReader.hasNext()) {
+                entry = dataFileReader.next();
+                recordCount++;
+            }
+            assertEquals(recordCount, 1);
+        }
+
+        // Alter data file entry to store incorrect file size
+        GenericData.Record dataFile = (GenericData.Record) entry.get("data_file");
+        long alteredValue = 50L;
+        assertNotEquals((long) dataFile.get("file_size_in_bytes"), alteredValue);
+        dataFile.put("file_size_in_bytes", alteredValue);
+        entry.put("data_file", dataFile);
```

Comment:
This shouldn't be needed since `get()` returns the original record object, which we already modified